packages-exp/auth-exp/test/mock_auth.ts

```diff
@@ -27,8 +29,29 @@ export const TEST_AUTH_DOMAIN = 'localhost';
 export const TEST_SCHEME = 'mock';
 export const TEST_KEY = 'test-api-key';
 
-export const mockAuth: Auth = new AuthImpl(
-  'test-app',
+export interface TestEnvironment {
+  auth: Auth,
+  persistence: MockPersistenceLayer,
+}
+
+export class MockPersistenceLayer extends InMemoryPersistence {
+  lastObjectSet: PersistedBlob | null = null;
+
+  set(key: string, object: PersistedBlob): Promise<void> {
+    this.lastObjectSet = object;
+    return super.set(key, object);
+  }
+
+  remove(key: string): Promise<void> {
+    this.lastObjectSet = null;
+    return super.remove(key);
+  }
+}
+
+export async function testEnvironment(): Promise<TestEnvironment> {
```

Comment:
this could be a lot simpler if we just had a persistence accessor on auth, this was a FR anyway and not a bad time to implement, then you wouldn't need this whole testenvironment construct

Reply:
Done-ish. I don't want to expose too many details of the persistence layer (there's the whole persistenceUserManager between the actual persistence and auth). Still, I've updated the auth object returned by the method to include the MockPersistence instance.